src/net/sourceforge/kolmafia/chat/ChatPoller.java

```diff
@@ -350,19 +351,19 @@ public static List<ChatMessage> parseNewChat(
       final boolean debug)
       throws JSONException {
     JSONArray msgs = obj.getJSONArray("msgs");
-    for (int i = 0; i < msgs.length(); i++) {
+    for (int i = 0; i < msgs.size(); i++) {
       JSONObject msg = msgs.getJSONObject(i);
 
       // If we have already seen this message in the chat GUI, skip it.
-      long mid = msg.optLong("mid");
+      long mid = msg.getLongValue("mid");
       if (!debug && mid != 0 && mid <= ChatPoller.serverLastSeen) {
         continue;
       }
 
       String type = msg.getString("type");
       boolean pub = type.equals("public");
 
-      String formatString = msg.optString("format");
+      String formatString = msg.getString("format");
```

Comment:
`optString` returns empty string if not present. I think this is a problem (/difference) in a lot of places.

Reply:
in this particular place I guess whoever wrote the code thought it returned null though!